src/main/java/com/isroot/stash/plugin/YaccConfigServlet.java

```diff
@@ -0,0 +1,180 @@
+
+        fields = new HashMap<String, String>();
+        fieldErrors = new HashMap<String, Iterable<String>>();
+    }
+
+
+    @SuppressWarnings("unchecked")
+    public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        log.debug("doGet");
+        settingsMap = (HashMap<String, Object>) pluginSettings.get(SETTINGS_MAP);
+        if (settingsMap == null) {
+            settingsMap = new HashMap<String, Object>();
+        }
+
+        validateSettings();
+        doGetContinue(req, resp);
+    }
+
+    public void validateSettings() {
+        Settings settings = repositoryHookService.createSettingsBuilder()
+                .addAll(settingsMap)
+                .build();
+        configValidator.validate(settings, new SettingsValidationErrorsImpl(fieldErrors), null);
+    }
+
+    protected void doGetContinue(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        log.debug("doGetContinue");
+        fields.clear();
+
+        for (Map.Entry<String, Object> entry : settingsMap.entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+            log.debug("got plugin config " + key + "=" + value + " " + value.getClass().getName());
+            if (value instanceof String) {
+                fields.put(key, (String) value);
+            }
+        }
+
+        log.debug("Config fields: " + fields);
+        log.debug("Field errors: " + fieldErrors);
+
+        resp.setContentType("text/html;charset=UTF-8");
+        try {
+            soyTemplateRenderer.render(resp.getWriter(), "com.isroot.stash.plugin.yacc:yaccHook-config-serverside", "com.atlassian.stash.repository.hook.ref.config",
+                    ImmutableMap
+                            .<String, Object>builder()
+                            .put("config", fields)
+                            .put("errors", fieldErrors)
+                            .build()
+            );
+        } catch (SoyException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof IOException) {
+                throw (IOException) cause;
+            }
+            throw new ServletException(e);
+        }
+
+    }
+
+    public void addStringFieldValue(Map<String, Object> settingsMap, HttpServletRequest req, String fieldName) {
+        String o;
+        o = req.getParameter(fieldName);
+        if (o != null && !o.isEmpty()) settingsMap.put(fieldName, o);
+    }
+
+    public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        settingsMap.clear();
+
+        for (Object key : req.getParameterMap().keySet()) {
+            String parameterName = (String) key;
+
+            // Plugin settings persister only supports map of strings
+            if (!parameterName.startsWith("errorMessage") && !parameterName.equals("submit")) {
```

Comment:
Looking better! Header/footer messages are now being saved. The other error messages are still not being saved.

> other custom error messages were correctly saved and repopulated (in Stash 3.10.2).

YACC currently supports Stash 2.12.0, which is the Stash version that runs when running `atlas-run`. I get the same behavior when running with Stash 3.0.0. It does work in 3.10.2 though (as you point out).

So, I think we need to figure out why it isn't work in all Stash versions... could you see if you can reproduce the issue using `atlas-run` and Stash 2.12.0 or 3.0.0? I am actually going to bump up the supported version to 3.0.0 because 2.x usage is very low. We can't bring it up to 3.10.2 though because 70% of YACC users are using Stash <=3.9.